### PR TITLE
Microservice dev

### PR DESCRIPTION
**Pull request by geratorres**
Microservice implementation

### CODE REVIEW

File: management-app/src/main/java/com/geratorres/controllers/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        ModelAndView modelAndView = new ModelAndView("productForm");
+
+        Product product = new Product();
+        modelAndView.addObject("productForm", product);
+
+        return modelAndView;
+    }
+
+    @PostMapping("/create")
+    public String formPost(@ModelAttribute("productForm") @Validated Product newProduct,
```
Comment (jass-holguin) [inline management-app/src/main/java/com/geratorres/controllers/ProductController.java]: here you are using the  `@Validated` annotation, but to validate y your product you are using your own validator, is this annotation doing anything?

File: management-app/src/main/java/com/geratorres/services/products/ProductServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public ProductServiceImpl(RabbitTemplate jacksonRabbitTemplate, ProductQueuesProperties properties) {
+        this.jacksonRabbitTemplate = jacksonRabbitTemplate;
+        this.properties = properties;
+    }
+
+    @Override
+    public void insertProduct(Product product) throws Exception {
+        Response response;
+
+        log.info("Requested to insertProduct: " + product);
```
Comment (jass-holguin) [inline management-app/src/main/java/com/geratorres/services/products/ProductServiceImpl.java]: you are logging when the method is entering, but you should also log what was retrieved after calling the `convertSandAndReceive()`

File: microservice/src/main/java/com/geratorres/repositories/ProductRepository.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.geratorres.repositories;
+
+import com.geratorres.entities.ProductEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository("productRepository")
+public interface ProductRepository extends JpaRepository<ProductEntity, Integer>, ProductRepositoryCustom {
```
Comment (jass-holguin) [inline microservice/src/main/java/com/geratorres/repositories/ProductRepository.java]: me quede pensando en esto. Si estas implementado tu `ProductRepositoryCustom` e inyectando el `EntityManager`, no bastaria con eso? O sea, creo que no necesitas un `ProductRepository` que extienda de `JpaRepository` porque ya estas creando tu propia implementacion y puedes usar directamente tu `ProductRepositoryCustom` como tu objeto repositorio.

File: microservice/src/main/java/com/geratorres/services/ProductService.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    public Response getProducts() {
+        log.info("Products Requested");
+
+        Response response = new Response();
+
+        try {
+            List<? extends Product> products = productRepository.listProducts();
+            response.setProducts((List<Product>) products);
+            response.succeed();
+        } catch (Exception ex) {
```
Comment (jass-holguin) [inline microservice/src/main/java/com/geratorres/services/ProductService.java]: log any exceptions

File: microservice/src/test/java/com/geratorres/entities/ProductEntityTest.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    @Test
+    public void description() {
+        String description = "desc";
+        productEntity.setDescription(description);
+
+        assertEquals(description, productEntity.getDescription());
+    }
+
+    @Test
+    public void toString1() {
```
Comment (jass-holguin) [inline microservice/src/test/java/com/geratorres/entities/ProductEntityTest.java]: name, also remember to use more descriptive names when testing, it should describe what you are testing,

File: microservice/src/test/java/com/geratorres/services/ProductServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    @Test
+    public void insertProductInvalid() {
+        assertFalse(productService.insertProduct(productEntityInvalid).isSucceeded());
+    }
+
+    //@Test
+    public void getProducts() {
+        //Mockito.when(productService.getProducts()).thenReturn(responseSuccess);
```
Comment (jass-holguin) [inline microservice/src/test/java/com/geratorres/services/ProductServiceTest.java]: dont leave commented code in your final PR

File: management-app/src/main/java/com/geratorres/controllers/ProductController.java (hunk 1 above)
Comment (geratorres) [reply management-app/src/main/java/com/geratorres/controllers/ProductController.java]: I bound my validator with the framework, so if I remove this annotation the validation is passed

**Pull request merged**